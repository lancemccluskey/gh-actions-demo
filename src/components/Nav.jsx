/** @jsx jsx */
import { jsx } from 'theme-ui';
import Link from 'next/link';

const Nav = () => {
  console.log('Nav');
  return (
    <header sx={{ variant: 'containers.header' }}>
      <nav sx={{ variant: 'containers.nav' }}>
        <Link href="/">
          <a sx={{ variant: 'containers.aHome' }}>Home</a>
        </Link>
        <Link href="/facts">
          <a sx={{ variant: 'containers.a' }}> Facts</a>
        </Link>
        <Link href="/jokes">
          <a sx={{ variant: 'containers.a' }}> Jokes</a>
        </Link>
        <Link href="/pics">
          <a sx={{ variant: 'containers.a' }}> Pics</a>
        </Link>
        <Link href="/quotes">
          <a sx={{ variant: 'containers.a' }}> Quotes</a>
        </Link>
      </nav>
    </header>
  );
};

export default Nav;
