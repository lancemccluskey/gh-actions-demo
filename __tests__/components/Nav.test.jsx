import { render, screen } from '@testing-library/react';
import Nav from '../../src/components/Nav';

describe('<Nav />', () => {
  it('should render', () => {
    const { container } = render(<Nav />);

    expect(container).toMatchSnapshot();
  });

  it('should have correct links', () => {
    render(<Nav />);

    expect(screen.getByText('Facts')).toBeInTheDocument();
    expect(screen.getByText('Jokes')).toBeInTheDocument();
    expect(screen.getByText('Pics')).toBeInTheDocument();
    expect(screen.getByText('Quotes')).toBeInTheDocument();
  });
});
