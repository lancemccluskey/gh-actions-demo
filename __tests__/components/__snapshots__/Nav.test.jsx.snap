// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav /> should render 1`] = `
<div>
  <header
    class="css-6ihouv-Nav"
  >
    <nav
      class="css-6ihouv-Nav"
    >
      <a
        class="css-6ihouv-Nav"
      >
        Home
      </a>
      <a
        class="css-6ihouv-Nav"
      >
         Facts
      </a>
      <a
        class="css-6ihouv-Nav"
      >
         Jokes
      </a>
      <a
        class="css-6ihouv-Nav"
      >
         Pics
      </a>
      <a
        class="css-6ihouv-Nav"
      >
         Quotes
      </a>
    </nav>
  </header>
</div>
`;
